### PR TITLE
Do not back up subscription-manager-initial-setup-addon

### PR DESCRIPTION
**Pull request by r0x0d**
Right now the `subscription-manager-initial-setup-addon` package exists in CentOS Linux 8.4, but was removed from CentOS Linux 8.5. That is causing a conversion error when that package is present on a CentOS Linux 8.5 system because convert2rhel is unable to back it up for rollback purposes.

The simplest fix to this is not backing up the package when we remove it as the backup process uses `yumdownloader` and therefore, `yumdownloader` uses any enabled repos that are present on the system.

This primary scenario to cover:
1. I have CentOS Linux 8.4 and I install subscription-manager-initial-setup-addon (it's available in the default repos)
2. I update the system to 8.5 (and I still have the package installed)
3. I run convert2rhel and shouldn't fail on backing up the package as it's N/A in 8.5 repos anymore (it does fail now)

The secondary scenario to cover:
1. I have CentOS Linux 8.4 and I install subscription-manager-initial-setup-addon (it's available in the default repos)
2. I run convert2rhel to convert to RHEL 8.4 EUS and it correctly backs up the package as we hardcode CentOS Linux 8.4 repos in convert2rhel

Jira reference (If any): https://issues.redhat.com/browse/OAMG-6422
Bugzilla reference (If any): https://bugzilla.redhat.com/show_bug.cgi?id=2046292

Signed-off-by: Rodolfo Olivieri <rolivier@redhat.com>

### CODE REVIEW

File: convert2rhel/subscription.py
Comment (bocekm) [inline convert2rhel/subscription.py:230]:
```suggestion
            # The package listed in _SUBMGR_PKG_REMOVED_IN_CL_85 has been removed from CentOS Linux 8.5 and causes
            # conversion to fail if it's installed on that system because it's not possible to back it up.
```
I realized that aftyer we start supporting conversions of 8.4 the message would then be confusing.

**Review by bocekm — CHANGES_REQUESTED** [review]

**Review by SpyTec — DISMISSED** [review]
LGTM. Integration tests should be added/updated

**Review by bocekm — APPROVED** [review]
Approving, but as @SpyTec metioned, we need an integration test before merging.

**Review by danmyway — APPROVED** [review]
The test looks fine to me. Thank you!

File: plans/tier1.fmf
Comment (kokesak) [inline plans/tier1.fmf:214]: In tier1 we use `checks-after-conversion` as a final test. Please add your `removed-pkgs-centos-85` to a preparation phase.

**Review by kokesak — DISMISSED** [review]
The test looks ok, just please change the tier1 file.

Also I have been thinking that we do not necessary need a special test file. I think you can add the installation of the package to the `resolve-dependency` test (https://github.com/oamg/convert2rhel/blob/main/tests/integration/tier1/resolve-dependency/install_dependency_packages.py)
I'll leave it up to you where you want to have the test located.

File: plans/tier1.fmf
Comment (hosekadam) [reply plans/tier1.fmf:237]: I would rather leave it in separate folder because of the different repos, which are added in the ansible part.

Comment (hosekadam) [reply plans/tier1.fmf]: @kokesak Anyway, I've changed the final test, now it should be fine.

**Review by bocekm — DISMISSED** [review]

File: tests/integration/tier1/removed-pkgs-centos-85/ansible/install_removed_pkgs_from_centos_85_repos.yml
Comment (SpyTec) [inline tests/integration/tier1/removed-pkgs-centos-85/ansible/install_removed_pkgs_from_centos_85_repos.yml:14]:
```suggestion
        baseurl: https://vault.centos.org/$contentdir/8.4.2105/BaseOS/$basearch/os/
        enabled: no
        gpgcheck: no
    - name: Add Centos 8.4.2105 AppStream repo
      yum_repository:
        name: appstream-84
        description: CentOS Linux 8.4.2105 - AppStream
        baseurl: https://vault.centos.org/$contentdir/8.4.2105/AppStream/$basearch/os/
```

**Review by SpyTec — DISMISSED** [review]

**Comment by bocekm** [comment]
We're getting `HTTP error (500 - Internal Server Error): The system is unable to complete the requested transaction` in the integration tests. Rerunning.

File: convert2rhel/subscription.py
Comment (abadger) [inline convert2rhel/subscription.py:248]: We do have this check in `utils.remove_pkgs()` already: https://github.com/oamg/convert2rhel/blob/main/convert2rhel/utils.py#L382

Would we want to rely on that instead of checking here as well?

File: convert2rhel/subscription.py
Comment (r0x0d) [reply convert2rhel/subscription.py:236]: Yes, this is a CentOS 8.5 only problem. IIRC, it shouldn't happen in any other version. 

I will test this again to be sure, but should be a CL8.5 only

**Review by abadger — APPROVED** [review]
Provisionally approved.  I think the code looks good but this is modifying the way CentOS 8 is converted to RHEL8 and I know the integration tests for conversion to RHEL8 are currently blocked on the ubi8 repo not having python3-cloud-what.

Once those tests are passing, definite +1 from me.  We could also decide to merge now and deal with any bugs later but that's not a decision I'd want to make by myself.

**Comment by r0x0d** [comment]
Follow-up comment on this one: https://github.com/oamg/convert2rhel/pull/429#discussion_r870525567

We discovered that this package is missing not only on CentOS 8.5 repositories, but on Oracle Linux 7 and Oracle Linux 8 as well. 

The only repository available with this package is CentOS 7, thus, I've updated the code to check for this package in every version other than CentOS 7 as this is the only one available with the package.

File: convert2rhel/subscription.py
Comment (bocekm) [inline convert2rhel/subscription.py:241]:
```suggestion
    # The subscription-manager-initial-setup-addon package is available in CentOS Linux 8.4 AppStream repo, but was removed from CentOS Linux 8.5. That causes a conversion error when that package is present on a CentOS Linux 8.5 system because convert2rhel is unable to back it up for rollback purposes. As a fix, we remove the package without backing it up on CentOS Linux 8.5.
    # Related: https://bugzilla.redhat.com/show_bug.cgi?id=2046292
    if system_info.id == "centos" and system_info.version.major == 8 and system_info.version.minor == 5:
```

**Review by bocekm — CHANGES_REQUESTED** [review]

**Review by bocekm — APPROVED** [review]

**Comment by bocekm** [comment]
I went through in total 4 tests failing on CentOS/Oracle Linux 7. All of them seem to be "infra"-related errors:

1) 3x `https://yum.oracle.com/repo/OracleLinux/OL7/latest/x86_64/getPackage/oraclelinux-release-el7-1.0-17.el7.x86_64.rpm: [Errno 14] curl#35 - "Peer reports it experienced an internal error."`
2) 1x `HTTP error (500 - Internal Server Error): The system is unable to complete the requested transaction`

For the above and because the code change should change behavior only on CentOS Linux 8 and since the tests on CentOS/Oracle Linux 8 pass, I'm merging it.

**Pull request merged**
